BlackMarket/app/Protocols/Validatable+String.swift

```diff
@@ -0,0 +1,68 @@
+//
+//  Validatable+String.swift
+//  swiftui-base
+//
+//  Created by Leandro Higa on 05/09/2022.
+//
+
+import Foundation
+
+enum ValidationType {
+  case email
+  case nonEmpty
+  case numeric
+  case date(dateFormat: String)
+  case phone
+  case none
+  case custom(isValid: (String) -> Bool)
+}
+
+internal protocol Validatable {
+  func validates(_ validations: [ValidationType]) -> Bool
+}
+
+extension String: Validatable {
+  func validates(_ validations: [ValidationType]) -> Bool {
+    validations.allSatisfy { validation in
+      switch validation {
+      case .email:
+        return isEmailFormatted()
+      case .numeric:
+        return isInteger()
+      case .date(dateFormat: let dateFormat):
+        return isDateFormatted(dateFormat)
+      case .phone:
+        return isPhoneNumber()
+      case .none:
```

Comment:
Do we really need the none validation type? Seems like you could do validations: [.none, .none] or [] and it would be the same thing. For clarity, this API could consider that an empty array of validations is the absence of any validation.